api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirConditionDaoImpl.java

```diff
@@ -0,0 +1,288 @@
+import org.openmrs.module.fhir2.api.search.param.SearchParameterMap;
+import org.openmrs.module.fhir2.api.util.LocalDateTimeFactory;
+import org.openmrs.util.PrivilegeConstants;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Component;
+
+@Component
+@Setter(AccessLevel.PUBLIC)
+@OpenmrsProfile(openmrsPlatformVersion = "2.0.5 - 2.1.*")
+public class FhirConditionDaoImpl extends BaseFhirDao<Obs> implements FhirConditionDao<Obs> {
+	
+	@Qualifier("sessionFactory")
+	@Autowired
+	private SessionFactory sessionFactory;
+	
+	private final boolean isRetireable;
+	
+	private final boolean isVoidable;
+	
+	protected FhirConditionDaoImpl() {
+		this.isRetireable = Retireable.class.isAssignableFrom(typeToken.getRawType());
+		this.isVoidable = Voidable.class.isAssignableFrom(typeToken.getRawType());
+	};
+	
+	private LocalDateTimeFactory localDateTimeFactory;
+	
+	@Override
+	@Authorized(PrivilegeConstants.GET_OBS)
+	public Obs get(@Nonnull String uuid) {
+		return super.get(uuid);
+	}
+	
+	@Override
+	@Authorized(PrivilegeConstants.EDIT_OBS)
+	public Obs createOrUpdate(@Nonnull Obs newEntry) {
+		return super.createOrUpdate(newEntry);
+	}
+	
+	@Override
+	@Authorized(PrivilegeConstants.DELETE_OBS)
+	public Obs delete(@Nonnull String uuid) {
+		return super.delete(uuid);
+	}
+	
+	@Override
+	@Authorized(PrivilegeConstants.GET_OBS)
+	public List<String> getSearchResultUuids(@Nonnull SearchParameterMap theParams) {
+		Criteria criteria = sessionFactory.getCurrentSession().createCriteria(Obs.class);
+		criteria.add(eq("concept.conceptId", FhirConstants.CONDITION_OBSERVATION_CONCEPT_ID));
+		
+		DetachedCriteria detachedCriteria = DetachedCriteria.forClass(Obs.class);
+		Criteria detachedExecutableCriteria = detachedCriteria.getExecutableCriteria(sessionFactory.getCurrentSession());
+		
+		if (isVoidable) {
+			handleVoidable(detachedExecutableCriteria);
+		} else if (isRetireable) {
+			handleRetireable(detachedExecutableCriteria);
+		}
+		
+		setupSearchParams(detachedExecutableCriteria, theParams);
+		handleSort(detachedExecutableCriteria, theParams.getSortSpec());
+		
+		detachedCriteria.setProjection(Projections.property("uuid"));
+		
+		criteria.add(propertyIn("uuid", detachedCriteria));
+		criteria.setProjection(Projections.groupProperty("uuid"));
+		
+		@SuppressWarnings("unchecked")
+		List<String> results = criteria.list();
+		
+		return results;
+	}
+	
+	@Override
+	@Authorized(PrivilegeConstants.GET_OBS)
+	public List<Obs> getSearchResults(@Nonnull SearchParameterMap theParams, @Nonnull List<String> matchingResourceUuids,
+	        int firstResult, int lastResult) {
+		return super.getSearchResults(theParams, matchingResourceUuids, firstResult, lastResult);
+	}
+	
+	private Optional<Criterion> handleAgeByDateProperty(@Nonnull String datePropertyName, @Nonnull QuantityParam age) {
```

Comment:
It might be better to refactor things so this method is in `BaseDao` and can be used from both this class and the 2.2 and later implementation.

Reply:
sure , i did that